File: app/src/main/java/com/booknara/android/apps/patterns/activity/SimpleFragmentActivity.java
package com.booknara.android.apps.patterns.activity;

import android.app.FragmentManager;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.booknara.android.apps.patterns.R;
import com.booknara.android.apps.patterns.fragment.SimpleFragment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import static com.booknara.android.apps.patterns.Constants.IntentKey.ACTIVITY_TITLE;

public class SimpleFragmentActivity extends AppCompatActivity {
    private static final String TAG = "SimpleFragmentActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_simple_fragment);

        Intent intent = getIntent();
        String title = "";
        if (intent != null) {
            title = intent.getStringExtra(ACTIVITY_TITLE);
        }

        Toolbar toolbar = findViewById(R.id.toolbar_view);
        toolbar.setTitle(title);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        Fragment fragment = SimpleFragment.newInstance();
        setDetailContainer(fragment);
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    private void setDetailContainer(Fragment fragment) {
        if (fragment == null) {
            Log.w(TAG, "fragment is null");
            return;
        }

        FragmentManager fragmentManager = getFragmentManager();
        Log.v(TAG, "fragment count : " + fragmentManager.getBackStackEntryCount());

        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.add(R.id.simple_fragment, fragment, SimpleFragment.TAG);
        ft.commitAllowingStateLoss();
    }

}

File: app/src/main/java/com/booknara/android/apps/patterns/activity/SimpleFragmentActivity.kt
package com.booknara.android.apps.patterns.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.commit
import com.booknara.android.apps.patterns.Constants.IntentKey.ACTIVITY_TITLE
import com.booknara.android.apps.patterns.R
import com.booknara.android.apps.patterns.databinding.ActivitySimpleFragmentBinding
import com.booknara.android.apps.patterns.fragment.SimpleFragment

class SimpleFragmentActivity: AppCompatActivity() {

  lateinit var binding: ActivitySimpleFragmentBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    binding = DataBindingUtil
      .setContentView(this, R.layout.activity_simple_fragment)

    val title = intent?.getStringExtra(ACTIVITY_TITLE) ?: ""

    binding.toolbarView.title = title
    setSupportActionBar(binding.toolbarView)
    supportActionBar?.setDisplayHomeAsUpEnabled(true)
    supportActionBar?.setDisplayShowHomeEnabled(true)

    setDetailContainer(SimpleFragment.newInstance())
  }

  override fun onSupportNavigateUp(): Boolean {
    onBackPressed()
    return true
  }

  private fun setDetailContainer(fragment: Fragment) {
//        val ft = supportFragmentManager.beginTransaction()
//        ft.add(R.id.simple_fragment, fragment, SimpleFragment.TAG)
//        ft.commitAllowingStateLoss()

    supportFragmentManager.commit(allowStateLoss = true) {
      add(R.id.simple_fragment, fragment, SimpleFragment.TAG)
    }
  }

  companion object {
    const val TAG = "SimpleFragmentActivity"
    const val TITLE = "Activity & Fragment"
  }
}
